test/System.IdentityModel.Tokens.Jwt.Tests/JwtSecurityTokenConverterTests.cs

```diff
@@ -0,0 +1,177 @@
+                ValidateIssuer = false,
+                TokenDecryptionKey = KeyingMaterial.DefaultSymmetricEncryptingCreds_Aes256_Sha512_512.Key,
+                IssuerSigningKey = KeyingMaterial.JsonWebKeyRsa256SigningCredentials.Key
+            };
+
+            var result = handler.ValidateToken(jweTokenString, validationParameters);
+            var jweToken = result.SecurityToken as JsonWebToken;
+
+            result = handler.ValidateToken(jwsTokenString, validationParameters);
+            var jwsTokenFromString = result.SecurityToken as JsonWebToken;
+
+            var jwsTokenFromHeaderAndPayload = new JsonWebToken(
+                Default.PayloadString,
+                new Microsoft.IdentityModel.Json.Linq.JObject
+                {
+                    { JwtHeaderParameterNames.Alg, SecurityAlgorithms.Sha512  },
+                    { JwtHeaderParameterNames.Kid, Default.AsymmetricSigningKey.KeyId },
+                    { JwtHeaderParameterNames.Typ, JwtConstants.HeaderType }
+                }.ToString(Microsoft.IdentityModel.Json.Formatting.None));
+
+            return new TheoryData<JwtSecurityTokenConverterTheoryData>
+            {
+                new JwtSecurityTokenConverterTheoryData
+                {
+                    First = true,
+                    TestId = "JweToJwe",
+                    InputToken = jweToken,
+                    Validator = (token) =>
+                    {
+                        Assert.NotNull(token.InnerToken);
+
+                        foreach (var header in jweToken.Header.Claims((string)Default.PayloadDictionary[JwtRegisteredClaimNames.Iss]))
+                        {
+                            Assert.True(token.Header.ContainsKey(header.Type));
+                            var otherHeader = token.Header[header.Type];
+                            Assert.Equal(header.Value, otherHeader);
+	                    }
+
+                        foreach (var header in jweToken.InnerToken.Header.Claims((string)Default.PayloadDictionary[JwtRegisteredClaimNames.Iss]))
+                        {
+                            Assert.True(token.InnerToken.Header.ContainsKey(header.Type));
+                            var otherHeader = token.InnerToken.Header[header.Type];
+                            Assert.Equal(header.Value, otherHeader);
+                        }
+
+                        var jweTokenClaims = new List<Security.Claims.Claim>(jweToken.Claims).ToDictionary(c => c.Type, c => c.Value);
+                        var otherTokenClaims = new List<Security.Claims.Claim>(token.Claims).ToDictionary(c => c.Type, c => c.Value);;
+
+                        Assert.Equal(jweTokenClaims.Count, otherTokenClaims.Count);
+
+                        foreach (var claim in jweTokenClaims)
+                        {
+                            Assert.True(otherTokenClaims.ContainsKey(claim.Key));
+                            Assert.True(otherTokenClaims[claim.Key] == claim.Value);
```

Comment:
Would Assert.Equal be more suitable here (and the other two, similar assertions)?